Assets/Scripts/Minigames/CowardMiniGame/RipcordBehavior.cs

```diff
@@ -12,216 +12,243 @@
-        {
-            _currentLight = 0;
-            StartCoroutine(blinkingLight());
-            _lightDoOnce = false;
-        }
-        if (_numReleased == 1 && _gameStarted == true && _lightDoOnce == true)
+
+        // Make sure the score detection range is destroyed once the ripcord returns to the start position
+        if(_scoreDetectionRange != null)
         {
-            _listLights[0].SetActive(true);
-            _currentLight = 1;
-            StartCoroutine(blinkingLight());
-            _lightDoOnce = false;
+            Destroy(_scoreDetectionRange);
         }
-        if(_numReleased == 2 && _gameStarted == true && _lightDoOnce == true)
+    }
+
+    /// <summary>
+    /// When the player successfully scores
+    /// </summary>
+    private void OnScore()
+    {
+        if (_canScore)
         {
-            _listLights[1].SetActive(true);
-            _currentLight = 2;
-            StartCoroutine(blinkingLight());
-            _lightDoOnce = false;
+            _score++;
+
+            OnRipcordScore?.Invoke();
+            _successfulPulls.text = (_score).ToString();
         }
-        if (_numReleased == 3)
+
+        // Winning score
+        if (_score >= 3)
         {
-            _listLights[2].SetActive(true);
-            StopCoroutine(blinkingLight());
             _gears.SetActive(true);
             _successfulPulls.color = Color.green;
-            StopRipcordSteam();
-            //Destroy(this);
-        }
-        //Debug.Log(_numReleased);
+            StopGeneratorSteam();
 
+            DeactivateMinigame();
+        }
     }
 
     /// <summary>
     /// Stops the steam coming from the ripcord.
     /// </summary>
-    public void StopRipcordSteam()
+    public void StopGeneratorSteam()
     {
         _steam.Stop();
     }
 
-    IEnumerator PauseEPress()
-    {
-        yield return new WaitForSeconds(0.4f);
-        _switchDoOnce = true;
-    }
-    IEnumerator blinkingLight()
-    {
-        _listLights[_currentLight].SetActive(true);
-        yield return new WaitForSeconds(_lightSpeed);
-        _listLights[_currentLight].SetActive(false);
-        _lightDoOnce = true;
-    }
-    IEnumerator ReleaseWindow()
-    {
-        _hasReleased = true;
-        yield return new WaitForSeconds(0.2f);
-        _hasReleased = false;
-    }
-    IEnumerator FindPreview()
-    {
-        _preview = GameObject.FindWithTag("Release");
-        yield return _preview;
-    }
-    void OnTriggerEnter(Collider col)
+    void OnTriggerEnter(Collider collider)
     {
-        if (col.gameObject.tag == "Release" && _hasReleased == true)
+        if (_isMinigameActive)
         {
-            _numReleased++;
-            GameObject.Destroy(col.gameObject);
-        }
+            // Enable player interaction
+            if (collider.gameObject.tag == "Player")
+            {
+                _playerController.Interact.started += OnInteract;
+                TabbedMenu.Instance.ToggleInteractPrompt(true, "GRAB");
```

Comment:
It might be worth making this "GRAB" string a serialized field to allow design to edit what this prompt displays. This would be more in line with how the other interact prompts work.